dbcon/joblist/jlf_subquery.cpp

```diff
@@ -90,19 +90,38 @@ void getColumnValue(ConstantColumn** cc, uint64_t i, const Row& row, const strin
 
         case CalpontSystemCatalog::DECIMAL:
         case CalpontSystemCatalog::UDECIMAL:
-            data = row.getIntField(i);
-            oss << (data / IDB_pow[row.getScale(i)]);
-
-            if (row.getScale(i) > 0)
+            if (row.getColumnWidth(i) == datatypes::MAXDECIMALWIDTH)
             {
-                if (data > 0)
-                    oss << "." << (data % IDB_pow[row.getScale(i)]);
-                else if (data < 0)
-                    oss << "." << (-data % IDB_pow[row.getScale(i)]);
+                int128_t val128;
```

Comment:
I would recomend you to take a look at examples in my decimalToString patch to simplify the boilerplate code.